carta/session.py

```diff
@@ -417,6 +417,43 @@ def clear_raster_scaling_reference(self):
         """Clear the raster scaling reference."""
         self.call_action("clearRasterScalingReference")
 
+    # VIEWER MODES
+    @validate(Boolean())
+    def set_viewer_mode(self, multiple=True):
+        """
+        Switch between single-panel mode and multiple-panel mode.
+
+        Parameters
+        ----------
+        multiple : {0}
+            Whether to use multiple-panel mode. Default is ``True``.
+        """
+        self.call_action("widgetsStore.setImageMultiPanelEnabled", multiple)
+
+    def previous_page(self):
+        """Go to previous page in viewer."""
+        self.call_action("widgetsStore.onPreviousPageClick")
+
+    def next_page(self):
+        """Go to next page in viewer."""
+        self.call_action("widgetsStore.onNextPageClick")
+
+    @validate(Number(), Number())
+    def set_viewer_grid(self, columns=2, rows=2):
```

Comment:
this function sets the grid layout. We would also need specify how the grid layout is interpreted. In the settings dialog, we can set the multi-panel mode as dynamic or fixed. In this set_view_grid function, I suggest we add another keyword argument as mode and having options as "dynamic" or "fixed".

Reply:
In ab3d8af, this request is solved by adding  a `mode` argument in `set_viewer_grid`. The `mode` is allowed to set to `"dynamic"` and `"fixed"`; if it's not these two inputs, an warning will be printed out to imply the user to set correct input.